### PR TITLE
[hail/ptypes] Style updates to InferPType

### PR DESCRIPTION
**Pull request by tpoterba**
Remove unnecessary brackets, add helper function

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/InferPType.scala [hunk 1]
```diff
@@ -55,415 +51,349 @@ object InferPType {
...
-      case InsertFields(old, fields, fieldOrder) => {
-        InferPType(old, env)
+      case InsertFields(old, fields, fieldOrder) =>
+        infer(old)
         val tbs = coerce[PStruct](old.pType2)
 
-        val s = tbs.insertFields(fields.map(f =>  {
-          InferPType(f._2, env)
+        val s = tbs.insertFields(fields.map(f => {
+          infer(f._2, env)
```
Comment (akotlar) [inline hail/src/main/scala/is/hail/expr/ir/InferPType.scala]: env not needed

File: hail/src/main/scala/is/hail/expr/ir/InferPType.scala [hunk 2]
```diff
@@ -55,415 +51,349 @@ object InferPType {
...
         val fd = t.field(name).typ
         fd.setRequired(t.required && fd.required)
-      }
       case MakeTuple(values) => PTuple(true, values.map(v => {
-          InferPType(v._2, env)
-          v._2.pType2
-      }):_*)
-      case MakeArray(irs, t) => {
-        if (irs.length == 0) {
+        infer(v._2, env)
```
Comment (akotlar) [inline hail/src/main/scala/is/hail/expr/ir/InferPType.scala]: env not needed

File: hail/src/main/scala/is/hail/expr/ir/InferPType.scala [hunk 3]
```diff
@@ -55,415 +51,349 @@ object InferPType {
...
+        infer(contextsIR)
+        infer(globalsIR)
+        infer(bodyIR, env.bind(contextsName -> contextsIR._pType2, globalsName -> globalsIR._pType2))
 
         PCanonicalArray(bodyIR._pType2, contextsIR._pType2.required)
-      }
-      case ReadPartition(rowIR, codecSpec, rowType) => {
-        InferPType(rowIR, env)
+      case ReadPartition(rowIR, codecSpec, rowType) =>
+        infer(rowIR, env)
```
Comment (akotlar) [inline hail/src/main/scala/is/hail/expr/ir/InferPType.scala]: env not needed

File: hail/src/main/scala/is/hail/expr/ir/InferPType.scala [hunk 4]
```diff
@@ -55,415 +51,349 @@ object InferPType {
...
-        if (irs.length == 0) {
+      case MakeStream(irs, t) =>
+        if (irs.isEmpty) {
           PType.canonical(t, true).deepInnerRequired(true)
         }
 
-        PStream(getNestedElementPTypes(irs.map( theIR => {
-          InferPType(theIR, env)
+        PStream(getNestedElementPTypes(irs.map(theIR => {
+          infer(theIR, env)
```
Comment (akotlar) [inline hail/src/main/scala/is/hail/expr/ir/InferPType.scala]: env

File: hail/src/main/scala/is/hail/expr/ir/InferPType.scala [hunk 5]
```diff
@@ -55,415 +51,349 @@ object InferPType {
...
-      case InsertFields(old, fields, fieldOrder) => {
-        InferPType(old, env)
+      case InsertFields(old, fields, fieldOrder) =>
+        infer(old)
         val tbs = coerce[PStruct](old.pType2)
 
-        val s = tbs.insertFields(fields.map(f =>  {
-          InferPType(f._2, env)
+        val s = tbs.insertFields(fields.map(f => {
+          infer(f._2, env)
```
Comment (akotlar) [inline hail/src/main/scala/is/hail/expr/ir/InferPType.scala]: bah one more

File: hail/src/main/scala/is/hail/expr/ir/InferPType.scala (hunk 2 above)
Comment (akotlar) [reply hail/src/main/scala/is/hail/expr/ir/InferPType.scala]: one more

File: hail/src/main/scala/is/hail/expr/ir/InferPType.scala (hunk 5 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/InferPType.scala]: aha, my regex didn't have punctuation

**Comment by akotlar** [comment]
some inference assertion failures

**Comment by akotlar** [comment]
some inference assertion failures

**Comment by akotlar** [comment]
needs bump

**Pull request merged**